app/src/pages/inside/launchesPage/modals/deleteLaunchModal/deleteLaunchModal.jsx

```diff
@@ -0,0 +1,245 @@
+  },
+  deleteImportantLaunches: {
+    id: 'DeleteLaunchModal.deleteImportantLaunches',
+    defaultMessage: 'Delete Important Launches',
+  },
+  deleteImportantLaunch: {
+    id: 'DeleteLaunchModal.deleteImportantLaunch',
+    defaultMessage: 'Delete Important Launch',
+  },
+  deleteWithImportantLaunch: {
+    id: 'DeleteLaunchModal.deleteWithImportantLaunch',
+    defaultMessage: 'Delete with Important Launches',
+  },
+  deleteOnlyRegular: {
+    id: 'DeleteLaunchModal.deleteOnlyRegular',
+    defaultMessage: 'Delete only regular',
+  },
+  deleteWithImportantLaunchMessage: {
+    id: 'DeleteLaunchModal.deleteWithImportantLaunchMessage',
+    defaultMessage:
+      '<b>There is "{name}" launch marked as important.</b><br />Are you sure you want to delete these important launches also?',
+  },
+  deleteWithImportantLaunchesMessage: {
+    id: 'DeleteLaunchModal.deleteWithImportantLaunchesMessage',
+    defaultMessage:
+      'There are <b>{importantCount} launches marked as important</b> among the {totalCount} launches you are trying to delete. Are you sure you want to delete these important launches also?',
+  },
+  deleteImportantLaunchMessage: {
+    id: 'DeleteLaunchModal.deleteImportantLaunchMessage',
+    defaultMessage:
+      '<b>”{name}” is marked as important.</b><br />Are you sure you want to delete this important launch?',
+  },
+  deleteImportantLaunchesMessage: {
+    id: 'DeleteLaunchModal.deleteImportantLaunchesMessage',
+    defaultMessage:
+      '<b>There are {importantCount} launches marked as important.</b><br />Are you sure you want to delete these important launches?',
+  },
+});
+
+const cx = classNames.bind(styles);
+
+const DeleteLaunchModal = ({ data: { launches, confirmDeleteLaunches, userId } }) => {
+  const { formatMessage } = useIntl();
+  const confirmAndClose = (closeModal) => {
+    confirmDeleteLaunches(launches);
+    closeModal();
+  };
+
+  const okButton = {
+    text: formatMessage(COMMON_LOCALE_KEYS.DELETE),
+    danger: true,
+    onClick: confirmAndClose,
+  };
+  const cancelButton = {
+    text: formatMessage(COMMON_LOCALE_KEYS.CANCEL),
+  };
+
+  const selectedImportantLaunches = launches.filter(
+    (launch) => launch.retentionPolicy === RETENTION_POLICY.IMPORTANT,
+  );
+
+  const selectedRegularLaunches = launches.filter(
+    (launch) => launch.retentionPolicy === RETENTION_POLICY.REGULAR,
+  );
+
+  const getMainContent = () => {
+    if (selectedImportantLaunches.length > 0 && selectedRegularLaunches.length > 0) {
+      return formatMessage(
+        selectedImportantLaunches.length === 1
+          ? messages.deleteWithImportantLaunchMessage
+          : messages.deleteWithImportantLaunchesMessage,
+        {
+          b: (data) => DOMPurify.sanitize(`<b>${data}</b>`),
+          name: selectedImportantLaunches[0].name,
+          importantCount: selectedImportantLaunches.length,
+          totalCount: launches.length,
+        },
```

Comment:
Let's split the messages formatting and put the only params that required for particular of them.